storage/range.go

```diff
@@ -149,6 +152,7 @@ type Range struct {
 	Desc      *proto.RangeDescriptor
 	rm        RangeManager  // Makes some store methods available
 	splitting int32         // 1 if a split is underway; updated atomically
+	lastIndex uint64        // last index in the raft log; updated atomically
```

Comment:
The lifecycle of a log entry is [unstable, stable, committed, applied]. lastIndex is the last stable entry; the synchronous processing you're talking about is from committed to applied. The last stable index and the last committed/applied index will definitely differ; this is where the quorum round-trip happens. I'll update the comment.